lib/nib/history/compose.rb

```diff
@@ -0,0 +1,48 @@
+require 'tempfile'
+require 'tmpdir'
+
+class Nib::History::Compose
+  attr_reader :dir
+
+  def initialize
+    @dir = "#{Dir.tmpdir}/#{Dir.pwd.split('/').last}"
+
+    FileUtils.mkdir_p(dir)
+  end
+
+  def path
+    file.path
+  end
+
+  private
+
+  def config
+    @config ||= YAML.safe_load(`docker-compose config`)
+  end
+
+  def file
+    @file ||= Tempfile.open('compose', dir) do |compose|
```

Comment:
`docker-compose` will create an image based on the project and service name (ie. `foo_web`). If `Tempdir.open` doesn't receive the second argument it will create the file in a random directory causing `docker-compose` to re-build the image (ie. `random_web`).